tests/browser/index.test.js

```diff
@@ -310,6 +310,35 @@ suite('<Entity events/>', () => {
     });
   });
 
+  test('can remove event listener when component is detached', function (done) {
+    let bazHandlerCalled = false;
+    function bazHandler () { bazHandlerCalled = true }
+
+    ReactDOM.render(
+      <Scene>
+        <Entity id='foo'>
+          <Entity id='bar' events={{baz: bazHandler}}/>
+        </Entity>
+      </Scene>,
+      div
+    );
+    div.querySelector('a-scene').addEventListener('loaded', () => {
+      ReactDOM.render(
+        <Scene>
+          <Entity id='foo'/>
+        </Scene>,
+        div
+      );
+      div.querySelector('a-entity').emit('baz');
```

Comment:
1. Entity `#bar` mounts, event handler is bound to `#bar`
2. Entity `#bar` unmounts 
3. Entity `#quuz` mounts without any event handler
4. Event is emitted on `#quuz`
5. Event handler is called on `#quuz`